src/fixup.js

```diff
@@ -297,6 +297,43 @@
     request.send();
   }
 
+  /* Dark mode toggle */
+  
+  const darkCss = document.querySelector('link[href="dark.css"]');
+  if (darkCss && matchMedia("(prefers-color-scheme)").matches) {
+
+    const toggleWrapper = document.createElement('div');
+    toggleWrapper.id = 'theme-toggle';
+    toggleWrapper.setAttribute('role', 'radiogroup');
+    toggleWrapper.setAttribute('aria-label', 'Select a color scheme');
+
+    document.body.dataset.colorScheme = localStorage.getItem("tr-theme") || "auto";
+
+    const createOption = (value) => {
+        const label = document.createElement('label');
+        const radio = document.createElement('input');
+        radio.type = 'radio';
+        radio.value = value;
+        radio.name = 'color-scheme';
+        radio.checked = value === document.body.dataset.colorScheme;
+        radio.addEventListener('change', event => {
+          document.body.dataset.colorScheme = event.target.value;
+          localStorage.setItem("tr-theme", event.target.value);
+        });
+        const text = document.createElement('span');
+        text.innerText = value;
+        label.appendChild(radio);
+        label.appendChild(text);
+        return label;
+    }
+
+    toggleWrapper.appendChild(createOption('light'));
+    toggleWrapper.appendChild(createOption('dark'));
+    toggleWrapper.appendChild(createOption('auto'));
+
+    document.querySelector('div.head').appendChild(toggleWrapper);
+  }
```

Comment:
React-hipster-wanna-be-version: 

```suggestion
  /* Dark mode toggle */
  const darkCss = document.querySelector('link[href="dark.css"]');
  if (darkCss && matchMedia("(prefers-color-scheme)").matches) {
    const render = document.createElement("x-temp");
    function createOption(option) {
      const checked = option === document.body.dataset.colorScheme;
      return `
        <label>
          <input name="color-scheme" type="radio" value="${option}"checked="${checked}">
          <span>${option}</span>
        </label>
      `.trim();
    }
    render.innerHTML = `
      <div id="theme-toggle" role="radiogroup" aria-label="Select a color scheme">
        ${["light", "dark", "auto"].map(createOption).join("")}
      </div>
    `;
    const changeListener = (event) => {
      const { value } = event.target;
      document.body.dataset.colorScheme = value;
      localStorage.setItem("tr-theme", value);
    };
    render.querySelectorAll("input[type='radio']").forEach((input) => {
      input.addEventListener("change", changeListener);
    });
    document.querySelector("div.head").append(...render.children);
  }
```

Reply:
Thanks for the suggestion. All looks great except that:

1. it's missing the local storage check at the beginning `document.body.dataset.colorScheme = localStorage.getItem("tr-theme") || "auto";`
2. the `checked` attribute l.308 should only be present for one and only one input.

Reply:
@marcoscaceres I've included your suggesting in https://github.com/w3c/tr-design/pull/243/commits/101f56929151e563ee39cc8edabbf13d8d1a413d